dozer-sql/src/pipeline/aggregation/sum.rs

```diff
@@ -1,208 +1,110 @@
-use crate::pipeline::aggregation::aggregator::Aggregator;
+use hashbrown::HashMap;
+use num_traits::FromPrimitive;
+use dozer_core::errors::ExecutionError::InvalidOperation;
+use dozer_types::ordered_float::OrderedFloat;
+use dozer_types::rust_decimal::Decimal;
+use dozer_types::tonic::codegen::Body;
+use crate::pipeline::aggregation::aggregator::{Aggregator, update_map};
 use crate::pipeline::errors::PipelineError;
 use dozer_types::types::{Field, FieldType};
+use crate::pipeline::expression::aggregate::AggregateFunctionType::Sum;
 
-// use crate::pipeline::aggregation::aggregator::AggregationResult;
-// use crate::pipeline::errors::PipelineError;
-// use crate::pipeline::errors::PipelineError::InvalidOperandType;
-// use crate::{deserialize, deserialize_decimal, deserialize_f64, deserialize_i64, deserialize_u64};
-// use dozer_core::storage::prefix_transaction::PrefixTransaction;
-// use dozer_types::ordered_float::OrderedFloat;
-// use dozer_types::rust_decimal::Decimal;
-// use dozer_types::types::{Field, FieldType};
-// use std::ops::{Add, Sub};
-//
-pub struct SumAggregator {}
+pub struct SumAggregator {
+    current_state: HashMap<Field, u64>,
```

Comment:
Why do  we need to keep an HashMap for a sum ? It would be enough to keep a value and add or remove a value from it

Reply:
but then we don't know what FieldType we are getting